File: frontend/src/objects/battle/microservices/session/objects/command/BattleCommandReason.ts
export type BattleCommandReason = "battle_finished" | "battle_inactive" | "player_dead" | "stale_command";

export function isBattleCommandReason(value: unknown): value is BattleCommandReason {
  return (
    value === "battle_finished" ||
    value === "battle_inactive" ||
    value === "player_dead" ||
    value === "stale_command"
  );
}

File: backend/src/main/scala/services/battle/microservices/runtime/objects/command/BattleCommandReason.scala
package services.battle.microservices.runtime.objects.command

enum BattleCommandReason {
  case BattleFinished
  case BattleInactive
  case PlayerDead
  case StaleCommand
}

object BattleCommandReason {
  def wireValue(value: BattleCommandReason): String =
    value match {
      case BattleCommandReason.BattleFinished => "battle_finished"
      case BattleCommandReason.BattleInactive => "battle_inactive"
      case BattleCommandReason.PlayerDead     => "player_dead"
      case BattleCommandReason.StaleCommand   => "stale_command"
    }

  def fromWire(value: String): Option[BattleCommandReason] =
    value match {
      case "battle_finished" => Some(BattleCommandReason.BattleFinished)
      case "battle_inactive" => Some(BattleCommandReason.BattleInactive)
      case "player_dead"     => Some(BattleCommandReason.PlayerDead)
      case "stale_command"   => Some(BattleCommandReason.StaleCommand)
      case _                 => None
    }
}
